draft-vandijk-dprive-ds-dot-signal-and-pin/yardsticks/draft-ietf-dprive-phase2-requirements-01.md

```diff
@@ -0,0 +1,116 @@
+This document quotes draft-ieft-dprive-phase2-requirements-01 to mark each 'requirement' as satisfied by the draft, unsatisfied by the draft, or commented otherwise. Every 'Yes' should be taken as 'when deployed by the relevant operators', or in other words 'the draft protocol permits it and the operator can choose to deploy'.
+
+> DNS Privacy Requirements for Exchanges between Recursive Resolvers and Authoritative Servers
+>
+>     draft-ietf-dprive-phase2-requirements-01
+> 
+> Abstract
+> 
+>    This document provides requirements for adding confidentiality to DNS
+>    exchanges between recursive resolvers and authoritative servers.
+> 
+> 5.  Requirements
+> 
+>    The requirements of different interested stakeholders are outlined
+>    below.
+> 
+> 5.1.  Mandatory Requirements
+> 
+>    1.   Each implementing party should be able to independently take
+>         incremental steps to meet requirements without the need for
+>         close coordination (e.g. loosely coupled)
+
+Yes.
+
+>    2.   Use a secure transport protocol between a recursive resolver and
+>         authoritative servers
+
+Yes.
+
+>    3.   Use a secure transport protocol between a recursive resolver and
+>         TLD servers
+
+Yes.
+
+>    4.   Use a secure transport protocol between a recursive resolver and
+>         the root servers
+
+Yes, but the quite unique situation of the root servers being operated by 13 parties might prove interesting when counting DS records.
```

Comment:
As long as you use the same hash algorithm adding more DNS records should not have a significant performance impact since the byte for byte comparison of a DS record does not add significant processing weight.

Also the thought occured to me that we could use CDS records for bootstrapping the root TLS pins. This would obviously be plain text (but secure due to the trust anchor), however it doesn't leak information beyond that you are using DoT which using port 853 does anyway.

Reply:
Text updated with your CDS idea.